test failures on conda-smithy feedstock

In https://github.com/conda-forge/conda-smithy-feedstock/pull/391, there's test failures, though only with rattler-build

<details>
<summary>List of failures</summary>

```
=========================== short test summary info ============================
FAILED tests/test_configure_feedstock.py::test_noarch_skips_appveyor[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_noarch_runs_on_circle[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_noarch_runs_on_default[rattler-build-recipe] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_noarch_runs_on_default[rattler-build-custom_recipe_dir] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_r_skips_appveyor[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_r_matrix_travis[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_r_matrix_on_circle[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_r_matrix_default[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_py_matrix_appveyor[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_py_matrix_travis[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_py_matrix_on_circle[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_py_matrix_on_github[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_py_matrix_on_azure[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_no_python_min_if_not_present[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_abi3_bools[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_upload_on_branch_azure[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_upload_on_branch_github_actions[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_upload_on_branch_appveyor[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_circle_osx[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_circle_skipped[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_render_windows_with_skipped_python[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_secrets[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_migrator_recipe[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_migrator_cfp_override[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_migrator_delete_old[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_migrator_downgrade_recipe[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_migrator_compiler_version_recipe[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_choco_install_azure[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_choco_install_github_actions[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_cos7_env_render[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_cuda_enabled_render[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_github_actions_populated[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_github_actions_labels[rattler-build-None] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_github_actions_labels[rattler-build-default] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_github_actions_labels[rattler-build-hosted] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_github_actions_labels[rattler-build-cirun-linux] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_github_actions_labels[rattler-build-self-hosted@another-linux-runner] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha[rattler-build-False-False-github_actions] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha[rattler-build-False-False-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha[rattler-build-False-True-github_actions] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha[rattler-build-False-True-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha[rattler-build-True-False-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha[rattler-build-True-True-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure[rattler-build-False-False-azure] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure[rattler-build-False-False-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure[rattler-build-False-True-azure] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure[rattler-build-False-True-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure[rattler-build-True-False-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure[rattler-build-True-True-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha_conditions[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_azure_conditions[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_overlapping_conditions[rattler-build-azure] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_overlapping_conditions[rattler-build-github_actions] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_store_build_artifacts_gha_and_azure_conditions[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-None] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-default] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-hosted] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-blacksmith-8vcpu-windows-2025] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-windows-2022] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-windows-2025] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha[rattler-build-namespace-profile-16cpu-on-win-64] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_azure[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_azure_old_vars[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_azure_old_and_new_vars[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_tools_dir[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_both[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-tools_install_dir-~/foo-specifies Unix path for Windows workflows: ~\\\\foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-tools_install_dir-foo-specifies Unix path for Windows workflows: foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-tools_install_dir-C:\\foo-specifies Windows path for Unix workflows: C:\\foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-tools_install_dir-C:\\\\foo-specifies Windows path for Unix workflows: C:\\\\foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-build_workspace_dir-~/foo-specifies Unix path for Windows workflows: ~\\\\foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-build_workspace_dir-foo-specifies Unix path for Windows workflows: foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-build_workspace_dir-C:\\foo-specifies Windows path for Unix workflows: C:\\foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_tools_build_paths_gha_override_wrong[rattler-build-build_workspace_dir-C:\\\\foo-specifies Windows path for Unix workflows: C:\\\\foo] - AssertionError: Regex pattern did not match.
FAILED tests/test_configure_feedstock.py::test_docker_run_args_gha[rattler-build-None-False] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_gha[rattler-build-None-True] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_gha[rattler-build---cap-add SYS_ADMIN-False] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_gha[rattler-build---cap-add SYS_ADMIN-True] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_azure[rattler-build-False-False] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_azure[rattler-build-False-True] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_azure[rattler-build-True-False] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_docker_run_args_azure[rattler-build-True-True] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-0-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-0-azure] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-0-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-0-both] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-16-azure] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-16-workflow_settings] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_azure[rattler-build-16-both] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_gha[rattler-build-None] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_gha[rattler-build-0] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_pagefile_size_gha[rattler-build-16] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha_old[rattler-build-False-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha_old[rattler-build-True-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha_old[rattler-build-value2-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha_old[rattler-build-value3-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha_old[rattler-build-value4-max] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha[rattler-build-False-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha[rattler-build-False-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha[rattler-build-False-max] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha[rattler-build-True-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha[rattler-build-True-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_gha[rattler-build-True-max] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure_old[rattler-build-False-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure_old[rattler-build-True-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure_old[rattler-build-value2-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure_old[rattler-build-value3-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure_old[rattler-build-value4-max] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure[rattler-build-False-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure[rattler-build-False-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure[rattler-build-False-max] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure[rattler-build-True-skip] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure[rattler-build-True-quick] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_free_disk_space_azure[rattler-build-True-max] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_configure_feedstock.py::test_namespace_pagefile_label[rattler-build] - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
FAILED tests/test_lint_recipe.py::TestLinter::test_test_section - AssertionError: 'The recipe must have some tests.' not found in ['The home item is expected in the about section.', 'The license item is expected in the about section.', 'The summary item is expected in the about section.', 'The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.', 'The recipe must have a `build/number` section.', 'Recipe name has invalid characters. only lowercase alpha, numeric, underscores, hyphens and dots allowed', 'Package version is missing.']
= 116 failed, 5003 passed, 14 skipped, 14 deselected, 1 xfailed, 6 warnings in 85.68s (0:01:25) =
```

</details>

This was with
```
rattler-build:              0.66.1-ha759004_0              conda-forge
```
and also with 0.65. 

I was surprised that this happened, given that CI is green here. However, looking at the logs of the most recent CI run here (https://github.com/conda-forge/conda-smithy/commit/c771d9b52815e29bf5f52bebf502c085b3d62147), we see that we're running with
```
rattler-build                         0.62.0               ha759004_0            conda-forge
```

This is a bit surprising given that I can't see any constraint on rattler-build in our setup here.

The error looks to be simply something like trying to read an empty file as a JSON
```
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 324, in render_recipe
    recipes = metadata.render_recipes(variants)
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 213, in render_recipes
    raise e
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 209, in render_recipes
    metadata = json.loads(content)
  File "$PREFIX/lib/python3.14/json/__init__.py", line 352, in loads
    return _default_decoder.decode(s)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "$PREFIX/lib/python3.14/json/decoder.py", line 345, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/json/decoder.py", line 363, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

CC @wolfv @baszalmstra @jaimergp

<details>
<summary>Example stack trace</summary>

```
=================================== FAILURES ===================================
__________________ test_noarch_skips_appveyor[rattler-build] ___________________
Traceback (most recent call last):
  File "$PREFIX/lib/python3.14/site-packages/_pytest/runner.py", line 353, in from_call
    result: TResult | None = func()
                             ~~~~^^
  File "$PREFIX/lib/python3.14/site-packages/_pytest/runner.py", line 245, in <lambda>
    lambda: runtest_hook(item=item, **kwds),
            ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
    ~~~~~~~~~~~~~~^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
    yield
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
    ~~~~~~~~~~~~~~^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/_pytest/capture.py", line 900, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
    ~~~~~~~~~~~~~~^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/_pytest/skipping.py", line 268, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "$PREFIX/lib/python3.14/site-packages/_pytest/runner.py", line 179, in pytest_runtest_call
    item.runtest()
    ~~~~~~~~~~~~^^
  File "$PREFIX/lib/python3.14/site-packages/_pytest/python.py", line 1720, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "$PREFIX/lib/python3.14/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "$PREFIX/lib/python3.14/site-packages/_pytest/python.py", line 166, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "$SRC_DIR/tests/test_configure_feedstock.py", line 37, in test_noarch_skips_appveyor
    configure_feedstock.render_appveyor(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        jinja_env=jinja_env,
        ^^^^^^^^^^^^^^^^^^^^
        forge_config=noarch_recipe.config,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        forge_dir=noarch_recipe.recipe,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "$PREFIX/lib/python3.14/site-packages/conda_smithy/configure_feedstock.py", line 1807, in render_appveyor
    return _render_ci_provider(
        "appveyor",
    ...<11 lines>...
        return_metadata=return_metadata,
    )
  File "$PREFIX/lib/python3.14/site-packages/conda_smithy/configure_feedstock.py", line 1346, in _render_ci_provider
    metas = rattler_render(
        os.path.join(forge_dir, forge_config["recipe_dir"]),
    ...<4 lines>...
        channel_urls=channel_sources,
    )
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 361, in render
    metadata_tuples = render_recipe(
        recipe_dir,
        config=config,
        variants=variants,
    )
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 324, in render_recipe
    recipes = metadata.render_recipes(variants)
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 213, in render_recipes
    raise e
  File "$PREFIX/lib/python3.14/site-packages/rattler_build_conda_compat/render.py", line 209, in render_recipes
    metadata = json.loads(content)
  File "$PREFIX/lib/python3.14/json/__init__.py", line 352, in loads
    return _default_decoder.decode(s)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "$PREFIX/lib/python3.14/json/decoder.py", line 345, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "$PREFIX/lib/python3.14/json/decoder.py", line 363, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Captured stdout call -----------------------------
Adding in variants from /tmp/pytest-of-conda/pytest-0/test_noarch_skips_appveyor_rat0/recipe/default_config.yaml
Captured stderr call -----------------------------
 Processing recipe at path: /tmp/pytest-of-conda/pytest-0/test_noarch_skips_appveyor_rat0/recipe/recipe.yaml

 [2m╭─[0m Finding outputs from recipe
 [2m│[0m Loading variant config from: /tmp/tmpum4azhsy
 [2m│[0m [33m[1m⚠ warning [0mFound 'pin_run_as_build' in variant config - this is not supported and will be ignored
 [2m│[0m Found 1 variants
 [2m│[0m 
 [2m│[0m Build variant: python-noarch-test-1.0.0-pyh4616a5c_0
 [2m│[0m 
 [2m│[0m ╭─────────────────┬──────────╮
 [2m│[0m │ Variant         ┆ Version  │
 [2m│[0m ╞═════════════════╪══════════╡
 [2m│[0m │ target_platform ┆ "noarch" │
 [2m│[0m ╰─────────────────┴──────────╯
 [2m│[0m
 [2m╰───────────────────[0m (took 0 seconds)
Captured log call -------------------------------
INFO     conda_build.variants:variants.py:289 Adding in variants from /tmp/pytest-of-conda/pytest-0/test_noarch_skips_appveyor_rat0/recipe/default_config.yaml
```

</details>

I ran a local build of https://github.com/conda-forge/conda-smithy-feedstock/pull/391 on a linux box now, and I cannot reproduce the errors; I only get
```
=========================== short test summary info ============================
FAILED tests/test_lint_recipe.py::TestLinter::test_test_section - AssertionEr...
= 1 failed, 5118 passed, 14 skipped, 14 deselected, 1 xfailed, 6 warnings in 322.28s (0:05:22) =
```
which (while it also shows up in the feedstock CI and not here) is not very critical; just about a linter expectation mismatch.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -76,11 +76,11 @@ jobs:
 
         # export CONDA_BUILD=1
         # export CONDA_BUILD_STATE=TEST
-        # export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
+        # export RATTLER_BUILD_COLOR=always
       - name: run tests
         run: |
           rm -rf conda_smithy
-          export RATTLER_BUILD_COLOR=always
+          export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
           pytest -vvsx tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -76,10 +76,10 @@ jobs:
 
         # export CONDA_BUILD=1
         # export CONDA_BUILD_STATE=TEST
+        # export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
       - name: run tests
         run: |
           rm -rf conda_smithy
-          export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
           export RATTLER_BUILD_COLOR=always
           pytest -vvsx tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -81,7 +81,7 @@ jobs:
           export CONDA_BUILD_STATE=TEST
           export RATTLER_BUILD_COLOR=always
           export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
-          pytest -vvsx tests --cov conda_smithy --cov-report lcov --cov-report term-missing
+          pytest tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -30,6 +30,7 @@ jobs:
           create-args: >-
             coverage
             coveralls
+            conda-forge-tick>=2026.0
 
       - name: install conda-smithy
         run: |

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -28,10 +28,8 @@ jobs:
           micromamba-version: 1.5.12-0
           cache-environment: true
           create-args: >-
-            python=3.12
             coverage
             coveralls
-            conda-forge-tick>=2026.0
 
       - name: install conda-smithy
         run: |
@@ -74,9 +72,14 @@ jobs:
           test -f sdist-content/conda_smithy-*/conda_smithy/templates/github-actions.yml.tmpl
           rm -r sdist-content/
 
+      # we remove the local package and set some env vars to emulate the smithy feedstock
+      # CI env in the hopes that we catch errors here.
       - name: run tests
         run: |
           rm -rf conda_smithy
+          export CONDA_BUILD=1
+          export CONDA_BUILD_STATE=TEST
+          export RATTLER_BUILD_COLOR=always
           export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
           pytest -vvsx tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -74,12 +74,12 @@ jobs:
           test -f sdist-content/conda_smithy-*/conda_smithy/templates/github-actions.yml.tmpl
           rm -r sdist-content/
 
+        # export CONDA_BUILD=1
+        # export CONDA_BUILD_STATE=TEST
       - name: run tests
         run: |
           rm -rf conda_smithy
-          export CONDA_BUILD=1
           export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
-          export CONDA_BUILD_STATE=TEST
           export RATTLER_BUILD_COLOR=always
           pytest -vvsx tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -61,7 +61,7 @@ jobs:
           popd
           pip uninstall conda-smithy --yes
 
-          python -m pip install -v --no-build-isolation -e .
+          python -m pip install -v --no-build-isolation .
 
       - name: test data and templates files in sdist
         run: |
@@ -76,6 +76,7 @@ jobs:
 
       - name: run tests
         run: |
+          rm -rf conda_smithy
           pytest tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -77,7 +77,11 @@ jobs:
       - name: run tests
         run: |
           rm -rf conda_smithy
-          pytest tests --cov conda_smithy --cov-report lcov --cov-report term-missing
+          export CONDA_BUILD=1
+          export RATTLER_BUILD_ENABLE_GITHUB_INTEGRATION=true
+          export CONDA_BUILD_STATE=TEST
+          export RATTLER_BUILD_COLOR=always
+          pytest -vvsx tests --cov conda_smithy --cov-report lcov --cov-report term-missing
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -28,10 +28,10 @@ jobs:
           micromamba-version: 1.5.12-0
           cache-environment: true
           create-args: >-
-            python=3.11
+            python=3.12
             coverage
             coveralls
-            conda-forge-tick
+            conda-forge-tick>=2026.0
 
       - name: install conda-smithy
         run: |

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -74,9 +74,6 @@ jobs:
           test -f sdist-content/conda_smithy-*/conda_smithy/templates/github-actions.yml.tmpl
           rm -r sdist-content/
 
-        # export CONDA_BUILD=1
-        # export CONDA_BUILD_STATE=TEST
-        # export RATTLER_BUILD_COLOR=always
       - name: run tests
         run: |
           rm -rf conda_smithy

--- a/environment.yml
+++ b/environment.yml
@@ -16,7 +16,6 @@ dependencies:
   # # Part of some optional lint tests
   - conda-recipe-manager>=0.9
   - conda-souschef
-  - conda-forge-tick>=2026.0
   # Runtime dependencies
   - conda>=22.11.1
   - conda-build>=26.3.0

--- a/environment.yml
+++ b/environment.yml
@@ -39,6 +39,7 @@ dependencies:
   - pixi>=0.59.0
   - jsonschema
   - exceptiongroup
-  # py-rattler's API subject to change, pin to minor
+  # rattler's APIs are subject to change, pin to minor versions
   - py-rattler>=0.22,<0.23
   - rattler-build-conda-compat>=1.4.15,<2.0.0a0
+  - rattler-build>=0.66.0,<0.67.0a0

--- a/environment.yml
+++ b/environment.yml
@@ -17,8 +17,8 @@ dependencies:
   - conda-recipe-manager>=0.9
   - conda-souschef
   # Runtime dependencies
-  - conda>=4.2
-  - conda-build>=25.3.1
+  - conda>=22.11.1
+  - conda-build>=26.3.0
   - conda-package-handling>=1.9.0
   - jinja2
   - requests

--- a/environment.yml
+++ b/environment.yml
@@ -16,6 +16,7 @@ dependencies:
   # # Part of some optional lint tests
   - conda-recipe-manager>=0.9
   - conda-souschef
+  - conda-forge-tick>=2026.0
   # Runtime dependencies
   - conda>=22.11.1
   - conda-build>=26.3.0
@@ -41,4 +42,4 @@ dependencies:
   - exceptiongroup
   # py-rattler's API subject to change, pin to minor
   - py-rattler>=0.22,<0.23
-  - rattler-build-conda-compat>=1.4.12,<2.0.0a0
+  - rattler-build-conda-compat>=1.4.15,<2.0.0a0

--- a/news/update-ci-2588.rst
+++ b/news/update-ci-2588.rst
@@ -1,0 +1,24 @@
+**Added:**
+
+* <news item>
+
+**Changed:**
+
+* Updated the testing CI configuration and dependencies to match that of the feedstock. (#2588)
+
+**Deprecated:**
+
+* <news item>
+
+**Removed:**
+
+* <news item>
+
+**Fixed:**
+
+* Fixed testing bugs where paths to files in the package were computed assuming a development
+  installation layout. (#2588)
+
+**Security:**
+
+* <news item>

--- a/news/update-ci-2588.rst
+++ b/news/update-ci-2588.rst
@@ -5,6 +5,7 @@
 **Changed:**
 
 * Updated the testing CI configuration and dependencies to match that of the feedstock. (#2588)
+* Added a pin on ``rattler-build`` to the minor version to help avoid breakages. (#2588)
 
 **Deprecated:**
 

--- a/tests/test_configure_feedstock.py
+++ b/tests/test_configure_feedstock.py
@@ -2223,7 +2223,7 @@ def test_github_actions_pins():
     pkg_root = Path(conda_smithy.__file__).parent
     github_actions_template = pkg_root / "templates" / "github-actions.yml.tmpl"
     dependabot_inventory = (
-        Path(__file__)
+        Path(__file__).parent
         / ".."
         / ".github"
         / "workflows"

--- a/tests/test_configure_feedstock.py
+++ b/tests/test_configure_feedstock.py
@@ -2221,11 +2221,13 @@ def test_github_actions_pins():
     make this pass.
     """
     repo_root = Path(conda_smithy.__file__).parents[1]
-    github_actions_template = (
-        repo_root / "templates" / "github-actions.yml.tmpl"
-    )
+    github_actions_template = repo_root / "templates" / "github-actions.yml.tmpl"
     dependabot_inventory = (
-        Path(__file__) / ".." / ".github" / "workflows" / "_proxy-file-for-dependabot-tests.yml"
+        Path(__file__)
+        / ".."
+        / ".github"
+        / "workflows"
+        / "_proxy-file-for-dependabot-tests.yml"
     )
 
     def get_uses(path):

--- a/tests/test_configure_feedstock.py
+++ b/tests/test_configure_feedstock.py
@@ -15,6 +15,7 @@ import yaml
 from conftest import ConfigYAML
 from rattler_build_conda_compat.loader import parse_recipe_config_file
 
+import conda_smithy
 from conda_smithy import configure_feedstock
 from conda_smithy.configure_feedstock import (
     DEFAULT_PROVIDER,
@@ -2219,12 +2220,12 @@ def test_github_actions_pins():
     If Dependabot opens a PR against the proxy, just copy the new pins to the template to
     make this pass.
     """
-    repo_root = Path(__file__).parents[1]
+    repo_root = Path(conda_smithy.__file__).parents[1]
     github_actions_template = (
-        repo_root / "conda_smithy" / "templates" / "github-actions.yml.tmpl"
+        repo_root / "templates" / "github-actions.yml.tmpl"
     )
     dependabot_inventory = (
-        repo_root / ".github" / "workflows" / "_proxy-file-for-dependabot-tests.yml"
+        Path(__file__) / ".." / ".github" / "workflows" / "_proxy-file-for-dependabot-tests.yml"
     )
 
     def get_uses(path):

--- a/tests/test_configure_feedstock.py
+++ b/tests/test_configure_feedstock.py
@@ -2220,8 +2220,8 @@ def test_github_actions_pins():
     If Dependabot opens a PR against the proxy, just copy the new pins to the template to
     make this pass.
     """
-    repo_root = Path(conda_smithy.__file__).parents[1]
-    github_actions_template = repo_root / "templates" / "github-actions.yml.tmpl"
+    pkg_root = Path(conda_smithy.__file__).parent
+    github_actions_template = pkg_root / "templates" / "github-actions.yml.tmpl"
     dependabot_inventory = (
         Path(__file__)
         / ".."

--- a/tests/test_schema.py
+++ b/tests/test_schema.py
@@ -59,8 +59,7 @@ def test_schema_no_empty_properties_for_bot():
     """
     with (
         Path(conda_smithy.__file__)
-        .parents[1]
-        .joinpath("data/conda-forge.json")
+        .parent.joinpath("data/conda-forge.json")
         .open("r") as f
     ):
         schema = json.load(f)

--- a/tests/test_schema.py
+++ b/tests/test_schema.py
@@ -6,6 +6,7 @@ from pathlib import Path
 
 import pytest
 
+import conda_smithy
 from conda_smithy.schema import ConfigModel
 from conda_smithy.utils import get_yaml
 from conda_smithy.validate_schema import (
@@ -57,9 +58,9 @@ def test_schema_no_empty_properties_for_bot():
     is not in the properties object in this case.
     """
     with (
-        Path(__file__)
+        Path(conda_smithy.__file__)
         .parents[1]
-        .joinpath("conda_smithy/data/conda-forge.json")
+        .joinpath("data/conda-forge.json")
         .open("r") as f
     ):
         schema = json.load(f)

